app/public/js/localSprites.js

```diff
@@ -217,17 +253,88 @@ function addFireballContrailParticle(parent, decayFrames, parentPreScalingXSize,
+    if (fireballParticle.name === 'fireballContrailParticle') localSprites.push(fireballParticle); //WRONG: Should push to parent.contrailParticles, but then render.js should render things in that array. I don't know the syntax for that yet, I don't think.
+    if (fireballParticle.name === 'fireballDetonationParticle') {
+        var randomVX,
+        randomVY;
+        if (Math.random() < 0.5) {
+            randomVX = -(Math.random() * 50);
+        } else {
+            randomVX = Math.random() * 50;
+        }
+        if (Math.random() < 0.5) {
+            randomVY = -(Math.random() * 50);
+        } else {
+            randomVY = Math.random() * 50;
+        }
+        fireballParticle.vx = randomVX;     //BROKEN: I'm not seeing these detonation particles move how they should.
+        fireballParticle.vy = randomVY + 4;
+        localSprites.push(fireballParticle);
+    }
+}
+
+function addFireballDetonation(parent, numberOfFragments, parentPreScalingXSize, parentPreScalingYSize) {
+    var detonationParticles = [];
+    for (var i = 0; i < numberOfFragments; i++) {
+        var newParticle = addFireballParticle(parent, 30, parentPreScalingXSize, parentPreScalingYSize, 1);
+        detonationParticles.push(newParticle);
+    }
+    for (var j = 0; j < detonationParticles.length; j++) {
+
+    }
+}
+
+
+function addTriggerZone(left, top, width, height, target, spawnedObjectType, spawnedObjectXOffset, spawnedObjectYOffset, xForce, yForce, cooldownFrames) {
+    //spawnedObjectXOffset and ...YOffset are where opject spawns relative to the center of the trigger zone
+    //spawnedObjectType key: 0 = doesn't spawn anything, 1 = homing fireball
+    var hitBox = rectangle(left, top, width, height);
+    var frames = [
+        $.extend(rectangle(1 * 16, 0 * 16, 16, 16), {image: customBlocksAImage, hitBox}),
+        //$.extend(rectangle(1 * 8, 0 * 8, 8, 8), {image: fireballContrailAImage, hitBox}),
+        //$.extend(rectangle(2 * 8, 0 * 8, 8, 8), {image: fireballContrailAImage, hitBox}),
+        //$.extend(rectangle(3 * 8, 0 * 8, 8, 8), {image: fireballContrailAImage, hitBox}),
+        //$.extend(rectangle(4 * 8, 0 * 8, 8, 8), {image: fireballContrailAImage, hitBox}),
+    ];
+    var triggerZone = new SimpleSprite({frames}, left, top, 0, 0, 1, 1);
+    triggerZone.name = 'triggerZone',
+    triggerZone.hitBox = rectangle(left, top, width, height),
+    triggerZone.target = target,
+    triggerZone.cooldownFrames = cooldownFrames,   //50 per second, I think
+    triggerZone.cooldownTimer = 0,
+    triggerZone.xForce = xForce,
+    triggerZone.yForce = yForce,
+    triggerZone.spawnedObjectType = 1,
+    triggerZone.spawnedObjectXOffset = (left + width / 2) + spawnedObjectXOffset,
+    triggerZone.spawnedObjectYOffset = (top + width / 2) + spawnedObjectYOffset,
+    triggerZone.framesToLive = 32767;   //would like to have an easy way to make this infinite, like a -1.
+    localSprites.push(triggerZone);
 }
 
 
+
```

Comment:
This is an excessive amount of white space to commit. I imagine it might make it easier to add some new lines while editing, but I don't think we should commit a bunch of blank lines to the repo.